src/QMCHamiltonians/SOECPComponent.cpp

```diff
@@ -192,6 +211,64 @@ SOECPComponent::RealType SOECPComponent::calculateProjector(RealType r, const Po
   return std::real(pairpot);
 }
 
+void SOECPComponent::evaluateOneBodyOpMatrixContribution(ParticleSet& W,
+                                                         const int iat,
+                                                         const TWFFastDerivWrapper& psi,
+                                                         const int iel,
+                                                         const RealType r,
+                                                         const PosType& dr,
+                                                         std::vector<ValueMatrix>& mats_b)
+{
+  RealType sold              = W.spins[iel];
+  using ValueVector          = SPOSet::ValueVector;
+  const IndexType gid        = W.getGroupID(iel);
+  const IndexType sid        = psi.getTWFGroupIndex(gid);
+  const IndexType firstIndex = W.first(gid);
+  const IndexType thisIndex  = iel - firstIndex;
+  const IndexType numOrbs    = psi.numOrbitals(sid);
+
+  ValueVector temp_row, up_row, dn_row;
+  up_row.resize(numOrbs);
+  dn_row.resize(numOrbs);
+  temp_row.resize(numOrbs);
+
+  //buildQuadraturePointDeltaPositions
+  for (int j = 0; j < nknot_; j++)
+    deltaV_[j] = r * rrotsgrid_m_[j] - dr;
+
+  //calculate radial potentials
+  for (int ip = 0; ip < nchannel_; ip++)
+    vrad_[ip] = sopp_m_[ip]->splint(r);
+
+  for (int iq = 0; iq < nknot_; iq++)
+  {
+    W.makeMove(iel, deltaV_[iq]);
+    psi.getRowMSpinDecomposed(W, iel, up_row, dn_row);
+    RealType jratio = psi.evaluateJastrowRatio(W, iel);
+    W.rejectMove(iel);
```

Comment:
To me, any NLPP code paths that require accept and reject are big no no. It can always be reformulated without need of accept/reject.

Reply:
Yea I agree, but I wanted to get it working first.

Reply:
Actually, on second thought we aren't doing anything costly like updating the inverse or anything like that by accepting/rejecting moves. the TWFwrapper being used here just grabs the jastrow ratio at each quadrature point and grabs the SPOSet values at the quadrature points. So I'm not sure there is much benefit from moving to using VirtualParticles